packages/components/src/components/marketplace-search.tsx

```diff
@@ -70,57 +80,109 @@ export const MarketplaceSearch = ({
-          >
-            <CloseIcon />
-          </button>
-        </div>
+        <MarketplaceSearchInput
+          onChange={setQuery}
+          value={query}
+          placeholder={placeholder}
+          className="mt-4"
+        />
 
-        <div className="flex flex-wrap gap-10 py-6 lg:flex-nowrap">
-          {items && queryList ? (
-            <MarketplaceList
-              title={queryList.title}
-              items={items}
-              placeholder={renderQueryPlaceholder(queryList.placeholder, query)}
-              pagination={queryList.pagination}
-            />
-          ) : (
-            <>
-              <MarketplaceList {...primaryList} />
-              {secondaryList && <MarketplaceList {...secondaryList} />}
-            </>
-          )}
-        </div>
+        {items && queryList ? (
+          <MarketplaceList
+            title={queryList.title}
+            items={items}
+            placeholder={renderQueryPlaceholder(queryList.placeholder, query)}
+            pagination={queryList.pagination}
+            colorScheme={colorScheme}
+          />
+        ) : (
+          <MarketplaceSearchTabs
+            primaryList={primaryList}
+            secondaryList={secondaryList}
+            colorScheme={colorScheme}
+            className="mt-8"
+          />
+        )}
       </div>
     </section>
   );
 };
+
+function MarketplaceSearchInput({
+  onChange,
+  value,
+  placeholder,
+  className,
+}: {
+  onChange: (value: string) => void;
+  value: string;
+  placeholder: string;
+  className?: string;
+}) {
+  return (
+    <div className="border-b border-[--fg-60]">
+      <div className={cn('hive-focus-within flex items-center rounded px-2', className)}>
+        <SearchIcon className="text-[--fg-80]" />
+        <input
+          value={value}
+          type="search"
+          placeholder={placeholder}
+          onChange={event => onChange(event.currentTarget.value)}
+          className="ml-2 w-full border-0 bg-transparent py-2 font-medium text-[--fg] outline-none placeholder:text-[--fg-60] [&::-webkit-search-cancel-button]:[display:none]"
+        />
+        <button
+          onClick={() => onChange('')}
+          className="hive-focus flex size-6 items-center justify-center rounded-sm"
+        >
+          <CloseIcon className="size-5 text-[--fg-80]" />
+        </button>
+      </div>
+    </div>
+  );
+}
+
+function MarketplaceSearchTabs({
+  primaryList,
+  secondaryList,
+  colorScheme,
+  className,
+}: {
+  primaryList: IMarketplaceListProps;
+  secondaryList: IMarketplaceListProps | undefined;
+  colorScheme: 'green' | 'neutral';
+  className?: string;
+}) {
+  return (
+    <div className={cn(classNames.tabs, className)}>
+      <Tabs items={[primaryList.title, secondaryList?.title].filter(x => x != null)}>
+        <Tabs.Tab tabIndex={-1}>
+          <MarketplaceList {...primaryList} title={undefined} colorScheme={colorScheme} />
```

Comment:
```suggestion
          <MarketplaceList {...primaryList} colorScheme={colorScheme} />
```
it was marked as optional, so we can avoid this?

Reply:
Oh but I'm actually resetting it. It's part of `primaryList`, it's spread and then I clear it. I'll leave a comment.